### PR TITLE
Deprecate Probability field on Opportunity

### LINKED ISSUE #5735: [Timebox] Remove "Probability" Field from Opportunities
# Current Behavior

We have a "Probability" field on `Opportunities`.

CleanShot 2024-06-04 at 15 44 34

# Desired Behavior

Remove this field as it's not useful.

### PR DESCRIPTION
**Pull request by ijreilly**
Closes #5735.
The field probability on opportunity will -
- stop being created for new workspaces (after this PR is merged)
- have "isCustom" value set to true and be displayed as such in the settings (after this PR is merged + sync-metadata is run on workspace)
- still show in the views (all the time)

This field is deprecated as a standard field but not replaced by another one, so we are not adding the `(deprecated)` suffix in the label.

### CODE REVIEW

File: packages/twenty-server/src/engine/workspace-manager/workspace-sync-metadata/constants/standard-field-ids.ts [hunk 1]
```diff
@@ -251,7 +251,7 @@ export const OPPORTUNITY_STANDARD_FIELD_IDS = {
   name: '20202020-8609-4f65-a2d9-44009eb422b5',
   amount: '20202020-583e-4642-8533-db761d5fa82f',
   closeDate: '20202020-527e-44d6-b1ac-c4158d307b97',
-  probability: '20202020-69d4-45f3-9703-690b09fafcf0',
+  probability_deprecated: '20202020-69d4-45f3-9703-690b09fafcf0',
```
Comment (FelixMalfait) [inline packages/twenty-server/src/engine/workspace-manager/workspace-sync-metadata/constants/standard-field-ids.ts:254]: Should we use probabilityDeprecated instead to remain consistent?

File: packages/twenty-server/src/modules/opportunity/standard-objects/opportunity.workspace-entity.ts [hunk 2]
```diff
@@ -63,13 +64,14 @@ export class OpportunityWorkspaceEntity extends BaseWorkspaceEntity {
...
   @WorkspaceField({
-    standardId: OPPORTUNITY_STANDARD_FIELD_IDS.probability,
+    standardId: OPPORTUNITY_STANDARD_FIELD_IDS.probability_deprecated,
     type: FieldMetadataType.TEXT,
     label: 'Probability',
     description: 'Opportunity probability',
     icon: 'IconProgressCheck',
     defaultValue: "'0'",
   })
+  @WorkspaceIsDeprecated()
```
Comment (FelixMalfait) [inline packages/twenty-server/src/modules/opportunity/standard-objects/opportunity.workspace-entity.ts:74]: Could we move all deprecated fields at the bottom of files by convention?

File: packages/twenty-server/src/modules/opportunity/standard-objects/opportunity.workspace-entity.ts [hunk 3]
```diff
@@ -63,13 +64,14 @@ export class OpportunityWorkspaceEntity extends BaseWorkspaceEntity {
...
   probability: string;
```
Comment (FelixMalfait) [inline packages/twenty-server/src/modules/opportunity/standard-objects/opportunity.workspace-entity.ts:75]: should this also be renamed to probabilityDeprecated? (if we want to introduce a new probability field soon?)

Comment (ijreilly) [reply packages/twenty-server/src/modules/opportunity/standard-objects/opportunity.workspace-entity.ts:75]: Im not sure - I don't think we have planned to introduce a new probability field for now, so the vision is to consider this field custom, not to discourage its usage for now. its what I had shared [in the doc](https://docs.google.com/document/d/1BJrg-f5jbCB0LhrvpmrvLW83nAAXFcEMGZcRSRf93Ak/edit) (doc is still to be completed though and migrated to github). but open to rediscuss this

File: packages/twenty-server/src/engine/workspace-manager/workspace-sync-metadata/constants/standard-field-ids.ts (hunk 1 above)
Comment (ijreilly) [reply packages/twenty-server/src/engine/workspace-manager/workspace-sync-metadata/constants/standard-field-ids.ts:254]: oops sure, thanks

File: packages/twenty-server/src/modules/opportunity/standard-objects/opportunity.workspace-entity.ts (hunk 3 above)
Comment (FelixMalfait) [reply packages/twenty-server/src/modules/opportunity/standard-objects/opportunity.workspace-entity.ts]: Ok!

File: packages/twenty-server/src/engine/workspace-manager/workspace-sync-metadata/constants/standard-field-ids.ts (hunk 1 above)
Comment (FelixMalfait) [reply packages/twenty-server/src/engine/workspace-manager/workspace-sync-metadata/constants/standard-field-ids.ts]: I saw Address_old was this way also (and did a quick PR fix for database:reset, make sure this PR doesn't break it also), maybe still time to fix it for Address

Comment (ijreilly) [reply packages/twenty-server/src/engine/workspace-manager/workspace-sync-metadata/constants/standard-field-ids.ts]: oops yes still time thanks for catching this

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**